File: ACM/Python/WFUStudy/P1469.py
'''
Author: KasperFan && fanwlx@foxmail.com
Date: 2023-11-18 15:48:44
LastEditTime: 2023-11-21 14:54:57
FilePath: /Python/WFUStudy/P1469.py
describes: This file is created for learning Python to deal OJ problems.
Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved. 
'''
import sys, os


def input(): return sys.stdin.readline()


n: int
ans: int = 0
i = 0
s = ''

n = int(input())
fr = os.dup(0)
os.set_blocking(fr, False)
while i < n:
    c = os.read(fr, 1).decode('utf-8')
    if c in [" ", "\n"]:
        x = int(s)
        ans ^= x
        s = ""
        i += 1
    else:
        s += c

print(ans)

File: ACM/Python/WFUStudy/P1469.c
/*
 * @Author: KasperFan && fanwlx@foxmail.com
 * @Date: 2023-11-18 15:15:16
 * @LastEditTime: 2023-11-28 21:07:24
 * @FilePath: /Python/WFUStudy/P1469.c
 * @describes: This file is created for learning Python to deal OJ problems.
 * Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved.
 */
#include <stdio.h>

int len[(int)1e9 + 10];
int have_len[(int)1e9 + 10];
int all_lengs[(int)5e6 + 10];
int idx, n, i;

int main()
{
    scanf("%d", &n);
    while (n--)
    {
        scanf("%d", &i);
        if (!have_len[i])
        {
            have_len[i] = 1;
            all_lengs[idx++] = i;
        }
        len[i]++;
    }
    for (int k = 0; k < idx; k++)
    {
        if (len[all_lengs[k]] % 2)
        {
            printf("%d\n", all_lengs[k]);
            return 0;
        }
    }

    return 0;
}


void example(){
    gets();
    fgets(str, 12, stdin);
}